src/io/dataset_loader.cpp

```diff
@@ -974,6 +1002,22 @@ void DatasetLoader::ConstructBinMappersFromTextData(int rank, int num_machines,
 
   dataset->feature_groups_.clear();
   dataset->num_total_features_ = std::max(static_cast<int>(sample_values.size()), parser->NumFeatures());
+
+  /*
+  int num_col = static_cast<int>(sample_values.size());
+  std::vector<double*> sample_values_ptr(num_col);
```

Comment:
Useless code should be deleted.

Reply:
This is also for debugging. If you like the idea of keep debugging code, I'd move those code into a separarte function that would only be called when debugging is enabled.